modules/cas_cache/layer_upgrade.c

```diff
@@ -1078,6 +1152,68 @@ static int _cas_upgrade_restore_conf_flush(struct cas_properties *cache_props,
 	return result;
 }
 
+static int _cas_upgrade_restore_conf_promotion(struct cas_properties *cache_props,
+		ocf_cache_t cache)
+{
+	uint64_t promotion_type;
+	uint64_t nhit_insertion_threshold;
+	uint64_t nhit_trigger_threshold;
+
+	int result = 0;
+
+	result = cas_properties_get_uint(cache_props, PROMOTION_POLICY_STR,
+			&promotion_type);
+	if (result) {
+		printk(KERN_ERR OCF_PREFIX_SHORT "Couldn't retrieve promotion "
+				"policy type \n");
+		goto out;
+	}
+
+	result = cache_mngt_set_promotion_policy(cache, promotion_type);
+	if (result) {
+		printk(KERN_ERR OCF_PREFIX_SHORT "Couldn't set promotion policy\n");
+		goto out;
+	}
+
+	if (promotion_type == ocf_promotion_always)
```

Comment:
maybe != ocf_promotion_nhit ? would be correct after adding new PP.

Reply:
Done